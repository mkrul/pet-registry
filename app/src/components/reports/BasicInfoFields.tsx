import React from "react";
import { TextField, SelectChangeEvent } from "@mui/material";
import { IReportForm } from "../../types/Report";

const commonInputStyles = {
  backgroundColor: "white",
  "& .MuiOutlinedInput-root": {
    backgroundColor: "white"
  }
};

interface BasicInfoFieldsProps {
  formData: IReportForm;
  onInputChange?: (
    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement> | SelectChangeEvent
  ) => void;
  readOnly?: boolean;
}

export const BasicInfoFields: React.FC<BasicInfoFieldsProps> = ({
  formData,
  onInputChange,
  readOnly
}) => {
  return (
    <>
      <TextField
        label="Title"
        name="title"
        value={formData.title || ""}
        onChange={onInputChange}
        variant="outlined"
        fullWidth
        disabled={readOnly}
        sx={commonInputStyles}
      />

      <TextField
        label="Description"
        name="description"
        value={formData.description || ""}
        onChange={onInputChange}
        multiline
        rows={4}
        variant="outlined"
        fullWidth
        disabled={readOnly}
        sx={commonInputStyles}
      />

      <TextField
        label="Pet's Name (leave blank if not known)"
        name="name"
        value={formData.name || ""}
        onChange={onInputChange}
        variant="outlined"
        fullWidth
        disabled={readOnly}
        sx={commonInputStyles}
      />
    </>
  );
};
