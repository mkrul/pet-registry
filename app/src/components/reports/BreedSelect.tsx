import React from "react";
import { getBreedsBySpecies, Species } from "../../lib/reports/breedLists";

interface IBreedSelect {
  species: Species | "";
  value: string;
  onChange: (value: string) => void;
}

const BreedSelect: React.FC<IBreedSelect> = ({ species, value, onChange }) => {
  const breeds = getBreedsBySpecies(species || null);

  return (
    <div className="w-full">
      <select
        name="breed"
        value={value}
        onChange={e => onChange(e.target.value)}
        className={`w-full min-w-[100px] rounded-md border-gray-300 py-2 pl-3 pr-10 text-sm focus:border-blue-500 focus:outline-none focus:ring-blue-500 whitespace-nowrap ${
          !species ? "bg-gray-100 text-gray-400 cursor-not-allowed" : ""
        }`}
        disabled={!species}
      >
        <option value="">Breed</option>
        {breeds.map(breed => (
          <option key={breed} value={breed}>
            {breed}
          </option>
        ))}
      </select>
    </div>
  );
};

export default BreedSelect;
