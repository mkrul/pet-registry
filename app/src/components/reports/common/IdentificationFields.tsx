import React from "react";
import {
  FormControl,
  TextField,
  Select,
  MenuItem,
  RadioGroup,
  FormControlLabel,
  Radio
} from "@mui/material";
import { getGenderOptions } from "../../../lib/reports/genderList";
import speciesListJson from "../../../../../config/species.json";
import BreedSearch from "../../common/BreedSearch";
import { AddFieldButton } from "../../common/AddFieldButton";
import { AdditionalFieldSet } from "../../common/AdditionalFieldSet";
import { IdentificationFieldsProps } from "../../../types/Report";
import { commonInputStyles } from "../../../styles/commonStyles";
import { FormFieldError } from "../../common/FormFieldError";

export const IdentificationFields: React.FC<IdentificationFieldsProps> = ({
  formData,
  showBreed2,
  onInputChange,
  setShowBreed2,
  onBreed2Remove,
  isLoading,
  error,
  breedError
}) => {
  const genderOptions = getGenderOptions();
  const speciesOptions = speciesListJson.options;

  const createChangeEvent = (name: string, value: string) =>
    ({
      target: { name, value }
    }) as React.ChangeEvent<HTMLInputElement>;

  const handleFieldChange = (name: string) => (value: string) => {
    onInputChange(createChangeEvent(name, value));
  };

  const handleSpeciesChange = handleFieldChange("species");
  const handleBreedChange = handleFieldChange("breed1");
  const handleBreed2Change = handleFieldChange("breed2");
  const handleGenderChange = handleFieldChange("gender");

  const handleBreed2Remove = () => {
    onInputChange(createChangeEvent("breed2", ""));
    onBreed2Remove?.();
    setShowBreed2(false);
  };

  return (
    <div className="space-y-6">
      <div className="space-y-2">
        <label className="text-lg font-medium text-gray-900 mb-2">Microchip ID:</label>
        <span className="text-sm text-gray-500"> (Leave blank if not known)</span>
        <TextField
          data-testid="microchip-id-input"
          name="microchipId"
          value={formData.microchipId || ""}
          onChange={onInputChange}
          variant="outlined"
          fullWidth
          sx={commonInputStyles}
          disabled={isLoading}
        />
      </div>

      <div className="space-y-2">
        <label className="text-lg font-medium text-gray-900 mb-2">Gender:</label>{" "}
        <span className="text-sm text-gray-500"> (Leave blank if not known)</span>
        <FormControl fullWidth>
          <Select
            data-testid="gender-select"
            labelId="gender-label"
            id="gender"
            value={formData.gender || ""}
            onChange={e => handleGenderChange(e.target.value)}
            sx={commonInputStyles}
            disabled={isLoading}
            MenuProps={{
              PaperProps: {
                style: {
                  maxHeight: 200
                }
              }
            }}
          >
            {genderOptions.map((gender, index) => (
              <MenuItem key={index} value={gender} data-testid="gender-option">
                {gender}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
      </div>

      <div className="space-y-2">
        <label className="text-lg font-medium text-gray-900">
          Is the animal spayed or neutered?
        </label>
        <div>
          <RadioGroup
            name="altered"
            value={formData.altered === null ? undefined : formData.altered.toString()}
            onChange={e => {
              console.log("Radio value changed to:", e.target.value);
              onInputChange(e);
            }}
          >
            <FormControlLabel value="1" control={<Radio />} label="Yes" />
            <FormControlLabel value="0" control={<Radio />} label="No" />
            <FormControlLabel value="" control={<Radio />} label="I don't know" />
          </RadioGroup>
          <FormFieldError error={error} />
        </div>
      </div>
      <div className="space-y-2">
        <label className="text-lg font-medium text-gray-900">Species:</label>
        <div>
          <TextField
            select
            name="species"
            value={formData.species || ""}
            onChange={onInputChange}
            variant="outlined"
            fullWidth
            required
            disabled={isLoading}
            sx={commonInputStyles}
            error={!!error}
          >
            {speciesOptions.map(option => (
              <MenuItem key={option} value={option}>
                {option}
              </MenuItem>
            ))}
          </TextField>
          <FormFieldError error={error} />
        </div>
      </div>

      <div className="space-y-2">
        <label className="text-lg font-medium text-gray-900">Breed:</label>
        <div>
          <BreedSearch
            species={formData.species.toLowerCase() as "dog" | "cat"}
            value={formData.breed1}
            onChange={handleBreedChange}
            disabled={isLoading}
            excludeBreeds={formData.breed2 ? [formData.breed2] : []}
            required
            size="medium"
            hideLabel
            disableClearable
            error={!!breedError}
            data-testid="breed-search"
          />
          <FormFieldError error={breedError} />
        </div>

        {!showBreed2 && formData.breed1 && (
          <AddFieldButton
            onClick={() => setShowBreed2(true)}
            disabled={isLoading}
            label="ADD BREED"
            testId="add-breed-button"
          />
        )}
      </div>

      {showBreed2 && (
        <div className="mt-8">
          <div className="mb-6">
            <AdditionalFieldSet
              label="Second Breed:"
              onRemove={handleBreed2Remove}
              disabled={isLoading}
              testId="remove-breed-button"
            >
              <BreedSearch
                species={formData.species.toLowerCase() as "dog" | "cat"}
                value={formData.breed2 || ""}
                onChange={handleBreed2Change}
                disabled={isLoading}
                excludeBreeds={[formData.breed1]}
                size="medium"
                hideLabel
                disableClearable
              />
            </AdditionalFieldSet>
          </div>
        </div>
      )}
    </div>
  );
};
