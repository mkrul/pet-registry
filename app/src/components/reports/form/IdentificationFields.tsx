import React from "react";
import { FormControl, TextField, InputLabel, Select, MenuItem, Button } from "@mui/material";
import { Species } from "../../../lib/reports/breedList";
import { getGenderOptions } from "../../../lib/reports/genderList";
import speciesListJson from "../../../../../config/species.json";
import BreedSearch from "../../common/BreedSearch";
import { IdentificationFieldsProps } from "../../../types/Report";
import CloseIcon from "@mui/icons-material/Close";
import { commonInputStyles } from "../../../styles/commonStyles";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPlus } from "@fortawesome/free-solid-svg-icons";
import Tip from "../../common/Tip";

export const IdentificationFields: React.FC<IdentificationFieldsProps> = ({
  formData,
  showBreed2,
  onInputChange,
  onBreedChange,
  onBreed2Change,
  onSpeciesChange,
  onShowBreed2Change,
  isLoading
}) => {
  const genderOptions = getGenderOptions();
  const speciesOptions = speciesListJson.options;

  return (
    <div className="space-y-6">
      <div className="space-y-2">
        <label className="text-lg font-medium text-gray-900 mb-2">Microchip ID:</label>
        <span className="text-sm text-gray-500"> (Leave blank if not known)</span>
        <TextField
          data-testid="microchip-id-input"
          name="microchipId"
          value={formData.microchipId || ""}
          onChange={onInputChange}
          variant="outlined"
          fullWidth
          sx={commonInputStyles}
          disabled={isLoading}
        />
      </div>

      <div className="space-y-2">
        <label className="text-lg font-medium text-gray-900 mb-2">Gender:</label>{" "}
        <span className="text-sm text-gray-500"> (Leave blank if not known)</span>
        <FormControl fullWidth>
          <Select
            data-testid="gender-select"
            labelId="gender-label"
            id="gender"
            value={formData.gender || ""}
            onChange={onInputChange}
            sx={commonInputStyles}
            disabled={isLoading}
            MenuProps={{
              PaperProps: {
                style: {
                  maxHeight: 200
                }
              }
            }}
          >
            {genderOptions.map((gender, index) => (
              <MenuItem key={index} value={gender} data-testid="gender-option">
                {gender}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
      </div>

      <div className="space-y-2">
        <label className="text-lg font-medium text-gray-900 mb-2">Species:</label>
        <FormControl fullWidth>
          <Select
            data-testid="species-select"
            labelId="species-label"
            id="species"
            value={formData.species}
            onChange={e => onSpeciesChange(e.target.value as Species)}
            sx={commonInputStyles}
            disabled={isLoading}
            MenuProps={{
              PaperProps: {
                style: {
                  maxHeight: 200
                }
              }
            }}
          >
            {speciesOptions.map((species, index) => (
              <MenuItem key={`${species}-${index}`} value={species} data-testid="species-option">
                {species}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
      </div>

      <div className="space-y-2">
        <label className="text-lg font-medium text-gray-900">Breed(s):</label>
        <Tip>
          Breeds can be difficult to identify visually. If you are unsure of the animal's primary
          breed makeup, use your best guess or select "Mixed Breed" from the dropdown.
        </Tip>
        <BreedSearch
          species={formData.species.toLowerCase() as "dog" | "cat"}
          value={formData.breed1}
          onChange={breed => onBreedChange(breed)}
          disabled={isLoading}
          excludeBreeds={formData.breed2 ? [formData.breed2] : []}
          required
          size="medium"
          hideLabel
          disableClearable
        />
        {!showBreed2 && formData.breed1 && (
          <Button
            data-testid="add-breed-button"
            onClick={() => onShowBreed2Change(true)}
            disabled={isLoading}
            color="primary"
            variant="text"
            className="mt-2"
            sx={commonInputStyles}
          >
            <div className="flex items-center">
              <FontAwesomeIcon icon={faPlus} className="mr-2 mb-[3px]" />
              <span>ADD ANOTHER BREED</span>
            </div>
          </Button>
        )}
      </div>

      {showBreed2 && (
        <div className="space-y-2">
          <label className="text-lg font-medium text-gray-900 mb-2">Second Breed:</label>
          <div className="flex items-center gap-4">
            <div className="flex-grow">
              <BreedSearch
                species={formData.species.toLowerCase() as "dog" | "cat"}
                value={formData.breed2 || ""}
                onChange={breed => onBreed2Change(breed)}
                disabled={isLoading}
                excludeBreeds={[formData.breed1]}
                size="medium"
                hideLabel
                disableClearable
              />
            </div>
            <Button
              data-testid="remove-breed-button"
              onClick={() => onShowBreed2Change(false)}
              disabled={isLoading}
              color="error"
              variant="text"
              startIcon={<CloseIcon fontSize="medium" />}
              aria-label="Remove Breed"
              sx={commonInputStyles}
            >
              Remove
            </Button>
          </div>
        </div>
      )}
    </div>
  );
};
