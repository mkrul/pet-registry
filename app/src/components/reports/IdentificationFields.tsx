import React from "react";
import {
  FormControl,
  TextField,
  InputLabel,
  Select,
  MenuItem,
  SelectChangeEvent,
  Button
} from "@mui/material";
import { IReportForm } from "../../types/Report";
import { Species } from "../../lib/reports/breedLists";
import { getGenderOptions } from "../../lib/reports/genderLists";
import speciesListJson from "../../../../config/species.json";
import BreedSelect from "./BreedSelect";

interface IdentificationFieldsProps {
  formData: IReportForm;
  showBreed2: boolean;
  onInputChange: (
    e:
      | React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
      | SelectChangeEvent
      | { target: { name: string; value: boolean | null } }
  ) => void;
  onBreedChange: (breed: string) => void;
  onBreed2Change: (breed: string) => void;
  onSpeciesChange: (species: Species) => void;
  onShowBreed2Change: (show: boolean) => void;
  isLoading?: boolean;
}

const commonInputStyles = {
  backgroundColor: "white",
  "& .MuiOutlinedInput-root": {
    backgroundColor: "white"
  },
  "& .MuiSelect-select": {
    backgroundColor: "white"
  }
};

export const IdentificationFields: React.FC<IdentificationFieldsProps> = ({
  formData,
  showBreed2,
  onInputChange,
  onBreedChange,
  onBreed2Change,
  onSpeciesChange,
  onShowBreed2Change,
  isLoading
}) => {
  const genderOptions = getGenderOptions();
  const speciesOptions = speciesListJson.options;

  return (
    <>
      <TextField
        label="Microchip ID (leave blank if not known)"
        name="microchipId"
        value={formData.microchipId || ""}
        onChange={onInputChange}
        variant="outlined"
        fullWidth
        sx={commonInputStyles}
      />

      <FormControl fullWidth>
        <InputLabel id="gender-label">Gender (leave blank if not known)</InputLabel>
        <Select
          labelId="gender-label"
          id="gender"
          name="gender"
          value={formData.gender || ""}
          onChange={onInputChange}
          label="Gender"
          sx={commonInputStyles}
        >
          {genderOptions.map((gender, index) => (
            <MenuItem key={index} value={gender}>
              {gender}
            </MenuItem>
          ))}
        </Select>
      </FormControl>

      <FormControl fullWidth>
        <InputLabel id="species-label">Species</InputLabel>
        <Select
          labelId="species-label"
          id="species"
          name="species"
          value={formData.species}
          onChange={e => onSpeciesChange(e.target.value as Species)}
          label="Species"
          sx={commonInputStyles}
        >
          {speciesOptions.map((species, index) => (
            <MenuItem key={`${species}-${index}`} value={species}>
              {species}
            </MenuItem>
          ))}
        </Select>
      </FormControl>

      <BreedSelect
        species={formData.species as Species | ""}
        value={formData.breed1}
        onChange={onBreedChange}
        variant="form"
        label="Breed 1"
      />

      {!showBreed2 && formData.breed1 && (
        <Button
          onClick={() => onShowBreed2Change(true)}
          disabled={isLoading}
          color="primary"
          variant="text"
          className="mt-2"
        >
          + ADD ANOTHER BREED
        </Button>
      )}

      {showBreed2 && (
        <div className="flex items-center gap-2">
          <BreedSelect
            species={formData.species as Species | ""}
            value={formData.breed2 || ""}
            onChange={onBreed2Change}
            variant="form"
            label="Breed 2"
            excludeBreeds={[formData.breed1]}
          />
          <button
            type="button"
            onClick={() => onShowBreed2Change(false)}
            className="text-red-600 hover:text-red-700 p-1 ml-1"
            disabled={isLoading}
            aria-label="Remove Breed 2"
          >
            âœ•
          </button>
        </div>
      )}
    </>
  );
};
