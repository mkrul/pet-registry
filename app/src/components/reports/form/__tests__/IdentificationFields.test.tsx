import { describe, it, expect, vi, beforeEach } from "vitest";
import { render, screen, act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { IdentificationFields } from "../IdentificationFields";
import { Species } from "../../../../lib/reports/breedList";
import type { ReportPropsForm } from "../../../../types/Report";
import { getGenderOptions } from "../../../../lib/reports/genderList";

describe("IdentificationFields", () => {
  const defaultProps = {
    formData: {
      microchipId: "",
      gender: "",
      species: "Dog" as Species,
      breed1: "",
      breed2: "",
      title: "",
      description: "",
      name: "",
      color1: "",
      color2: "",
      color3: "",
      dateLastSeen: "",
      address: "",
      city: "",
      state: "",
      zipCode: "",
      images: [],
      image: {
        id: "0",
        url: "",
        filename: "",
        thumbnailUrl: "",
        variantUrl: "",
        publicId: ""
      },
      area: "",
      country: "",
      latitude: 0,
      longitude: 0
    } as ReportPropsForm,
    showBreed2: false,
    onInputChange: vi.fn(),
    onBreedChange: vi.fn(),
    onBreed2Change: vi.fn(),
    onSpeciesChange: vi.fn(),
    onShowBreed2Change: vi.fn(),
    isLoading: false
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("renders correctly", () => {
    const { container } = render(<IdentificationFields {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it.each([
    { input: "123456789", expectedCalls: 9 },
    { input: "ABCD1234", expectedCalls: 8 }
  ])("handles microchip ID input changes: $input", async ({ input }) => {
    const user = userEvent.setup();
    render(<IdentificationFields {...defaultProps} />);

    const microchipInput = screen.getByTestId("microchip-id-input").querySelector("input");
    expect(microchipInput).toBeDefined();

    await act(async () => {
      if (microchipInput) {
        await user.type(microchipInput, input);
      }
    });

    expect(defaultProps.onInputChange).toHaveBeenCalledTimes(input.length);
  });

  it.each(getGenderOptions())("handles gender selection for %s", async gender => {
    const user = userEvent.setup();
    render(<IdentificationFields {...defaultProps} />);

    await act(async () => {
      const genderSelect = screen.getByTestId("gender-select");
      await user.click(genderSelect);
    });

    // Wait for MUI menu to appear
    const menu = document.querySelector('[role="listbox"]');
    expect(menu).toBeDefined();

    if (menu) {
      const options = menu.querySelectorAll('[role="option"]');
      const option = Array.from(options).find(opt => opt.textContent === gender);
      expect(option).toBeDefined();

      if (option) {
        await act(async () => {
          await user.click(option);
        });

        expect(defaultProps.onInputChange).toHaveBeenCalledWith(
          expect.objectContaining({
            target: { name: "gender", value: gender }
          })
        );
      }
    }
  });

  it.each(["Cat", "Dog"])("handles species selection for %s", async species => {
    const user = userEvent.setup();
    render(<IdentificationFields {...defaultProps} />);

    await act(async () => {
      const speciesSelect = screen.getByTestId("species-select");
      await user.click(speciesSelect);
    });

    // Wait for MUI menu to appear
    const menu = document.querySelector('[role="listbox"]');
    expect(menu).toBeDefined();

    if (menu) {
      const options = menu.querySelectorAll('[role="option"]');
      const option = Array.from(options).find(opt => opt.textContent === species);
      expect(option).toBeDefined();

      if (option) {
        await act(async () => {
          await user.click(option);
        });
        expect(defaultProps.onSpeciesChange).toHaveBeenCalledWith(species);
      }
    }
  });

  it("shows add another breed button when breed1 is selected", () => {
    render(
      <IdentificationFields
        {...defaultProps}
        formData={{ ...defaultProps.formData, breed1: "Labrador" }}
      />
    );

    const addBreedButton = screen.getByTestId("add-breed-button");
    expect(addBreedButton).toBeDefined();
  });

  it("shows breed2 field when showBreed2 is true", () => {
    render(
      <IdentificationFields
        {...defaultProps}
        showBreed2={true}
        formData={{ ...defaultProps.formData, breed1: "Labrador" }}
      />
    );

    // Use queryByTestId instead of getByTestId to avoid throwing
    const breed2Field = screen.queryByTestId("breed-search-2");
    const removeButton = screen.queryByTestId("remove-breed-button");
    expect(breed2Field).toBeDefined();
    expect(removeButton).toBeDefined();
  });

  it("handles remove breed2 button click", async () => {
    const user = userEvent.setup();
    render(
      <IdentificationFields
        {...defaultProps}
        showBreed2={true}
        formData={{ ...defaultProps.formData, breed1: "Labrador" }}
      />
    );

    await act(async () => {
      const removeButton = screen.getByTestId("remove-breed-button");
      await user.click(removeButton);
    });

    expect(defaultProps.onShowBreed2Change).toHaveBeenCalledWith(false);
  });

  it("disables all interactive elements when isLoading is true", () => {
    render(
      <IdentificationFields
        {...defaultProps}
        isLoading={true}
        formData={{ ...defaultProps.formData, breed1: "Labrador" }}
      />
    );

    // Check button disabled state
    const addBreedButton = screen.getByTestId("add-breed-button");
    expect(addBreedButton).toBeDisabled();

    // For MUI TextField, check the input element
    const microchipInput = screen.getByTestId("microchip-id-input").querySelector("input");
    expect(microchipInput).toBeDisabled();

    // For MUI Select components
    const genderSelect = screen.getByTestId("gender-select");
    const speciesSelect = screen.getByTestId("species-select");

    [genderSelect, speciesSelect].forEach(select => {
      expect(select).toHaveClass("Mui-disabled");
    });

    // For BreedSearch components
    const breedSearchControl = screen.getByTestId("breed-search-form-control");
    const autocompleteInput = breedSearchControl.querySelector('input[role="combobox"]');
    expect(autocompleteInput).toBeDisabled();

    // If breed2 is shown, check that one too
    const breed2SearchControl = screen.queryByTestId("breed-search-form-control");
    if (breed2SearchControl) {
      const autocompleteInput2 = breed2SearchControl.querySelector('input[role="combobox"]');
      if (autocompleteInput2) {
        expect(autocompleteInput2).toBeDisabled();
      }
    }

    // Check remove breed button if visible
    const removeButton = screen.queryByTestId("remove-breed-button");
    if (removeButton) {
      expect(removeButton).toBeDisabled();
    }
  });
});
