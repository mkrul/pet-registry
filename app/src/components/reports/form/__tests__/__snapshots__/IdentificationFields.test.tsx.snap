// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IdentificationFields > renders correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-17uadx6-MuiFormControl-root-MuiTextField-root"
    data-testid="microchip-id-input"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for=":r0:"
      id=":r0:-label"
    >
      Microchip ID (leave blank if not known)
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        name="microchipId"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-81qg8w"
        >
          <span>
            Microchip ID (leave blank if not known)
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      id="gender-label"
    >
      Gender (leave blank if not known)
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-10mr75f-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="gender-select"
    >
      <div
        aria-controls=":r1:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="gender-label gender"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="gender"
        role="combobox"
        tabindex="0"
      >
        <span
          aria-hidden="true"
          class="notranslate"
        >
          â€‹
        </span>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
        name="gender"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-81qg8w"
        >
          <span>
            Gender
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      id="species-label"
    >
      Species
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-10mr75f-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="species-select"
    >
      <div
        aria-controls=":r2:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="species-label species"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="species"
        role="combobox"
        tabindex="0"
      >
        Dog
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
        name="species"
        tabindex="-1"
        value="Dog"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w1u3ce"
        >
          <span>
            Species
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
    data-testid="breed-search-form-control"
  >
    <div
      class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon css-1tlcqt-MuiAutocomplete-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-12bc0o0-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r3:"
          id=":r3:-label"
        >
          Breed
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1n04w30-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r3:"
            required=""
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
          <div
            class="MuiAutocomplete-endAdornment css-1uhhrmm-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-nykp2v-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
              tabindex="-1"
              title="Clear"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </button>
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-1puztro-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-81qg8w"
            >
              <span>
                Breed
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
